runtime/android/core/src/org/xwalk/core/XWalkActivity.java

```diff
@@ -231,22 +232,37 @@ protected void dismissDialog() {
     }
 
     private void getXWalkLibrary() {
-        try {
-            Intent intent = new Intent(Intent.ACTION_VIEW);
-            startActivity(intent.setData(Uri.parse(XWALK_CORE_MARKET_URL)));
-            showDialog(getMarketProgressDialog());
-        } catch (ActivityNotFoundException e) {
-            Log.d(TAG, "Market open failed");
-            showDialog(getMarketOpenFailedDialog());
-            // TODO(sunlin): enable direct download in the future
-            // downloadXWalkLibrary();
```

Comment:
Commit log is good for this PR review. But think about when other people reading this piece of code, it may modified as well, it is hard to identify which commit log should be read to get the logic.